core/text/table/table.odin

```diff
@@ -37,10 +49,19 @@ Table :: struct {
 	dirty: bool, // True if build() needs to be called before rendering
 
 	// The following are computed on build()
-	colw: [dynamic]int, // Width of each column (including padding, excluding borders)
+	colw: [dynamic]int, // Width of each column (excluding padding and borders)
 	tblw: int,          // Width of entire table (including padding, excluding borders)
 }
 
+ascii_width_proc :: proc(str: string) -> int {
+	return len(str)
+}
+
+unicode_width_proc :: proc(str: string) -> (width: int) {
+	_, _, width = #force_inline utf8.grapheme_count(str)
```

Comment:
`utf8.grapheme_count()` calls `utf8.decode_grapheme_clusters()` which allocates implicitly using `context.allocator`. Isn't this a hidden allocation and a memory leak?
All allocators used should be specified up front at `init`.

Reply:
```
Inputs:
- str: The input string.
- track_graphemes: Whether or not to allocate and return `graphemes` with extra data about each grapheme.
- allocator: (default: context.allocator)

```

```
grapheme_count :: proc(str: string) -> (graphemes, runes, width: int) {
	_, graphemes, runes, width = decode_grapheme_clusters(str, false)
```

```
				if track_graphemes {
					append(&graphemes, Grapheme{
						byte_index,
						rune_count,
						width - last_width,
					})
				}
```

This interpretation is incorrect. In addition, `grapheme_count` is tested under the test suite, and if a memory leak occurs, it would show up in the warnings.

Reply:
I stand corrected. Ignore my comment.